### PR TITLE
Added Domain.Interfaces library

### PR DESCRIPTION
**Pull request by andreykalachev**
Also entities from Domain.Core library were moved  to Domain.Core/Entities folder, New DTO entities were added to Domain.Core/DTOs

### CODE REVIEW

File: Core/DTOs/IdAndNameDto.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
```
Comment (dmyhun-andersen) [inline Core/DTOs/IdAndNameDto.cs]: 1. Wrong name:
F.E.:
CategoryViewModel
CategoryVM
CategoryDTO
CategoryServiceModel

2. Wrong namespace and location:
where do these dto are going to be used?
They should be on the separate layer like OnionApp.Domain.ServiceModels or OnionApp.Api.ViewModels

File: Core/DTOs/RelationBasicInfoDTO.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace OnionApp.Domain.Core.DTOs
+{
+    public class RelationBasicInfoDto
```
Comment (dmyhun-andersen) [inline Core/DTOs/RelationBasicInfoDTO.cs]: The same with location

File: OnionApp.Domain.Interfaces/Repositories/ICategoryRepository.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using OnionApp.Domain.Core.DTOs;
+
+namespace OnionApp.Domain.Interfaces.Repositories
+{
+    public interface ICategoryRepository
+    {
+        Task<IEnumerable<IdAndNameDto>> GetBasicInfoAllAsync();
```
Comment (dmyhun-andersen) [inline OnionApp.Domain.Interfaces/Repositories/ICategoryRepository.cs]: It looks like ICategoryService

File: OnionApp.Domain.Interfaces/Repositories/ICountryRepository.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Collections.Generic;
+using System.Threading.Tasks;
+using OnionApp.Domain.Core.DTOs;
+
+namespace OnionApp.Domain.Interfaces.Repositories
+{
+    public interface ICountryRepository
+    {
+        Task<IEnumerable<IdAndNameDto>> GetBasicInfoAllAsync();
```
Comment (dmyhun-andersen) [inline OnionApp.Domain.Interfaces/Repositories/ICountryRepository.cs]: It looks like ICountryService

File: OnionApp.Domain.Interfaces/Repositories/IGenericRepository.cs [hunk 5]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+namespace OnionApp.Domain.Interfaces.Repositories
+{
+    public interface IGenericRepository<TEntity> where TEntity : class
+    {
+
```
Comment (dmyhun-andersen) [inline OnionApp.Domain.Interfaces/Repositories/IGenericRepository.cs]: remove empty line

File: OnionApp.Domain.Interfaces/Repositories/IGenericRepository.cs [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        
+        Task<TEntity> GetAsync(int id);
+
+        Task<TEntity> FindAsync(Expression<Func<TEntity, bool>> predicate);
+
+        Task<IEnumerable<TEntity>> GetAllAsync();
+
+        Task<IEnumerable<TEntity>> GetAllAsync(Expression<Func<TEntity, bool>> predicate);
+
+        void Add(TEntity entity);
```
Comment (dmyhun-andersen) [inline OnionApp.Domain.Interfaces/Repositories/IGenericRepository.cs]: Can we make AddAsync, DeleteAsync and UpdateAsync?

Comment (dmyhun-andersen) [reply OnionApp.Domain.Interfaces/Repositories/IGenericRepository.cs]: image

Comment (andreykalachev) [reply OnionApp.Domain.Interfaces/Repositories/IGenericRepository.cs]: I think there is no point in this. My understanding is that these methods will only change entity state and then we can commit changes async

**Pull request merged**